hack/libe2e.sh

```diff
@@ -122,3 +120,23 @@ function stdout_gt() {
     fi
     return 1
 }
+
+function dump_debug_logs() {
+    local namespace="${1}"
+    local output_dir="$(pwd)/_artifacts/${namespace}"
+    echo "Dumping cluster state to ${output_dir}"
+    mkdir -p "${output_dir}"
+    kubectl cluster-info dump --namespaces "${namespace}" > "${output_dir}/dump.txt" || true
```

Comment:
❓ I mentioned it in previous review, but it might be good to use the `--output-directory` option here, which will result in a directory with separate yaml files for each of the resource types and separate log files for each of the pods.